### PR TITLE
Add button to create note if not found

### LINKED ISSUE #1586: Create notes from 404 page
**Is your feature request related to a problem? Please describe.**
Right now it's not possible to create notes directly using a url.

**Describe the solution you'd like**
As mentioned in https://github.com/hedgedoc/hedgedoc/issues/1767 it would be nice to drop the workflow "if you access a free url and the note doesn't exist.. we create it!". So we should add a "Create this note" button on the 404 page.

**Describe alternatives you've considered**
Reimplement the "if you access a free url and the note doesn't exist.. we create it!" workflow.

### PR DESCRIPTION
**Pull request by mrdrogdrog**
### Component/Part
error message if note is not found

### Description
This PR adds a button on the error page to create a non-existent note

### Steps

- [x] Added implementation
- [x] I read the [contribution documentation](https://github.com/hedgedoc/react-client/blob/main/CONTRIBUTING.md) and signed-off my commits to accept the DCO.

### Related Issue(s)
fixes #1586

### CODE REVIEW

File: src/api/notes/index.ts [hunk 1]
```diff
@@ -51,6 +51,7 @@ export const createNote = async (markdown: string): Promise<Note> => {
  * @return Content and metadata of the new note.
  */
 export const createNoteWithPrimaryAlias = async (markdown: string, primaryAlias: string): Promise<Note> => {
+  await new Promise((resolve) => setTimeout(resolve, 5000))
```
Comment (ErikMichelson) [inline src/api/notes/index.ts:54]: Why did you add this 5 second delay?

File: src/components/common/note-loading-boundary/create-non-existing-note-hint.tsx [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+import { Alert, Button } from 'react-bootstrap'
+import { useSingleStringUrlParameter } from '../../../hooks/common/use-single-string-url-parameter'
+import { createNoteWithPrimaryAlias } from '../../../api/notes'
+import { useAsyncFn } from 'react-use'
+import { ShowIf } from '../show-if/show-if'
+import { Redirect } from '../redirect'
+import { ForkAwesomeIcon } from '../fork-awesome/fork-awesome-icon'
+import { testId } from '../../../utils/test-id'
+
+export const CreateNonExistingNoteHint: React.FC = () => {
```
Comment (ErikMichelson) [inline src/components/common/note-loading-boundary/create-non-existing-note-hint.tsx:18]: ESDoc missing

File: src/components/common/note-loading-boundary/create-non-existing-note-hint.tsx [hunk 3]
```diff
@@ -0,0 +1,76 @@
...
+      <Alert variant={'danger'} {...testId('failedMessage')}>
+        <ForkAwesomeIcon icon={'exclamation-triangle'} className={'mr-1'} />
+        <Trans i18nKey={'noteLoadingBoundary.createNote.error'} />
+      </Alert>
+    )
+  } else {
+    return (
+      <div className={'d-flex flex-column mt-5'}>
+        <b>
+          <Trans i18nKey={'noteLoadingBoundary.createNote.question'} values={{ aliasName: noteIdFromUrl }} />
```
Comment (ErikMichelson) [inline src/components/common/note-loading-boundary/create-non-existing-note-hint.tsx:57]: Is the content of `values` cached properly for re-renders, as this is an object and not a plain data type?

**Review by ErikMichelson — CHANGES_REQUESTED** [review]

File: src/api/notes/index.ts (hunk 1 above)
Comment (mrdrogdrog) [reply src/api/notes/index.ts:54]: Testing code

File: src/components/common/note-loading-boundary/create-non-existing-note-hint.tsx (hunk 3 above)
Comment (mrdrogdrog) [reply src/components/common/note-loading-boundary/create-non-existing-note-hint.tsx:57]: In this case it should be okay because this part of the component doesn't get rerendered that often.

Comment (ErikMichelson) [reply src/components/common/note-loading-boundary/create-non-existing-note-hint.tsx:61]: Okay. When we do performance measurements at some point, we could take a look again.

**Review by ErikMichelson — COMMENTED** [review]
It's only a style question.
But in my opinion the space between the question and the button is too small, considering the large space above the question.

image

Even more, when respecting that the error message doesn't have the large space.

image

**Comment by mrdrogdrog** [comment]
I increased the space between the alert and the headline and also between the button and the text.

The spacing is inteded this way to see which elements are grouped.

**Comment by mrdrogdrog** [comment]
I have another idea. Wait a second

**Comment by mrdrogdrog** [comment]
@ErikMichelson I changed the code to use alerts. This should give enough separation.

**Review by ErikMichelson — APPROVED** [review]
Looks much better!

**Review by DerMolly — APPROVED** [review]
